config/src/main/java/com/typesafe/config/impl/BeanFactory.java

```diff
@@ -0,0 +1,160 @@
+package com.typesafe.config.impl;
+
+import com.typesafe.config.Config;
+import com.typesafe.config.ConfigException;
+
+import java.beans.IntrospectionException;
+import java.beans.Introspector;
+import java.beans.PropertyDescriptor;
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * Sent as pull request to config project.
+ * https://github.com/typesafehub/config/pull/107 (original)
+ * https://github.com/typesafehub/config/pull/249 (automatic conversion support)
```

Comment:
Sorry, this is my first pull request. Can you explain what wrong with this lines?